### PR TITLE
Template domain name

### LINKED ISSUE #5: Look into replacing hard-coded domain ^^^
This is currently hard-coded in a lot of places. 

- [x] Replace in TF for DC and WEC
- [x] Replace in provisioning powershell scripts 
- [x] Replace in XML for GPOs
- [x] Replace in XML for WEF subscriptions
- [x] Test it's working...

### PR DESCRIPTION
**Pull request by alice-carr**
This PR templates the domain name

- [x] in TF for DC and WEC
- [x] in provisioning powershell scripts
- [x] in XML for GPOs
- [x] in XML for WEF subscriptions

### CODE REVIEW

**Comment by alice-carr** [comment]
Didn't get as far with this as I wanted to today: 
- I added the domain as a variable in TF and removed the hard coding in the terraform files.
- I looked into how we could do this in the powershell files but I haven't come to any conclusions just yet - I'm still working powershell out and I can see a bunch of `.ps1` files that include the domain name, but I can't quite figure out how these files are called / used. We can pass arguments in when calling powershell functions from the command line though.

File: terraform/modules/windows-test-domain/scripts/WEC/join_domain.ps1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
-$User = "shire.com\wecserver"
+$User = "$env:domain\wecserver"
 $Password = ConvertTo-SecureString -String "Edhellen$" -AsPlainText -Force
 $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $User, $Password
 
-Add-Computer -DomainName "shire.com" -OUPath "OU=Servers,DC=shire,DC=com" -Credential $Credential
+Add-Computer -DomainName "$env:domain" -OUPath "OU=Servers,DC=shire,DC=com" -Credential $Credential
```
Comment (danjoneslf) [inline terraform/modules/windows-test-domain/scripts/WEC/join_domain.ps1:5]: I think these `"OU=Servers,DC=shire,DC=com"` probably need to change but I'm not sure what to.

Comment (alice-carr) [reply terraform/modules/windows-test-domain/scripts/WEC/join_domain.ps1:5]: ah. Well spotted.

**Review by danjoneslf — COMMENTED** [review]
I wonder why it thinks those gpreport.xml files are binary?

File: terraform/modules/windows-test-domain/scripts/WinRM/templating.ps1 [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    $output = switch -regex -file $file.fullname {
+        $matchString { $_ -replace $matchString,$replaceString
+                       $matchFound = $true
+        }
+        default { $_ }
+    }
+    if ($matchFound) {
+        $output | Set-Content $file.fullname
+    }
+}
```
Comment (danjoneslf) [inline terraform/modules/windows-test-domain/scripts/WinRM/templating.ps1:15]: This is very minimalist - Nice!

**Review by danjoneslf — APPROVED** [review]
Lets merge this and then see what happens when we start invoking the join domain scripts.

**Pull request merged**